### PR TITLE
(Ready for Merge) Church code rework

### PR DESCRIPTION
**Pull request by SingingSpock**
## About The Pull Request
<details>
<summary>
	 Rework of litanies, linked to hopeful upcoming lore update (though not inherently tied to it, the PR does not depend upon the new lore)
</summary>
<hr>
Reworks the litanies of Church in a handful of ways. 

Scriptures taken from the Holy Bible, New International Version®, NIV®. Copyright © 1973, 1978, 1984, 2011 by Biblica, Inc.™ Used by permission of Zondervan. All rights reserved worldwide. www.zondervan.com The “NIV” and “New International Version” are trademarks registered in the United States Patent and Trademark Office by Biblica, Inc.®
<hr>
</details>

## Changelog
:cl:

- Changed a few defines to be actually readable rather than sticking to poor Eris practices. While it'll make porting things marginally more difficult, it's mostly just a find/replace job.
- The nutrition and blood costs were scrapped and replaced with a perk given by the cruciform which makes tool-related actions take 25% longer. This doesn't apply on tools that are Sanctified since the sanctifier already gives -50% workspeed
- Every single litany got at the very least a lookover of its code and its spoken text. Each text has been looked up, many of them altered, and all of them given a translation in a comment next to it. I am working on updating the Guide to Absolutism page to include all of these translations. Grammar errors have been fixed wherever I can find them.
- Tessellate and Factorial litanies that heal organs were fixed to work with Erismed. Factorial healing litanies were given a downside in the form of a 5 minute stat debuff (and a hardstun for the self-heal) to balance for the fact that they're near aheals and limit their use as mid-combat heals.
- Many little changes were made to litanies as well, too many to fully document. Some things fail louder or have had their numbers tweaked. Some litanies were moved to more accurate categories. Some had exploits fixed.
- Soul Hunger tweaked to put a mild slowing poison into the speaker rather than do burn damage. This poison can be removed quicker by drinking Cahors.
- Manifestation now checks if there's a dense object in the space when creating golems and machines. No more stacking five solidifiers in the same space. It also checks if you're already building something and prevents building more than one item at a time.
- The "Initiate" and "Devotion" categories were renamed to "Clerical" and "Episcopal" respectively because those names actually have meaning that relates to their function.
- Made the Episcopal litanies (the ones that either require 100 power or explicitly check for a Prime) only appear when using a Prime litany book. They can still be spoken from memory by anyone with enough power and the perms.
- Added litanies to A) Make someone a Vector but not a Prime (our current system only permits promotion to Prime directly), B) reduce a Vector/Prime to a lay state, and C) remove the cruciform of someone who's still alive. The latter two can only be done by a Prime.
- The Inquisitor now actually functions as the Church equivalent of the Marshal Special Agent, with an admin spawner and two special litanies designed to help them with their tasks. They're also the only people able to use reduction/separation litanies on Primes
- Made Crusaders actually use the crusader cruciform module again rather than the inquisitor module. The litanies may get a balance pass, but for now they've been left mostly unaltered.
- Minor tweaks to the Will of the Protector, mainly to distinguish for the players whether the Will gave Inspiration or did a Scan.
- Renamed the Sanctify ritual to Bless Land and make it turn tiles around the speaker holy, with a radius dependent on the number of participants. Lower radius in the deep maintenance (but it's more generally useful there)
- All the cloning stuff was moved to its own file which isn't included in the .dme file. Keeping it for archival is good, but it doesn't need to clutter the main files.
- The Nullifier was reworked to still do a very small amount of healing passively while inactive, and to do its healing in smaller chunks over 2 minute intervals rather than at the end of a 10 minute activation. Overall healing over the course of a 10 minute activation was reduced, but the passive even while inactive should make up for it.
- Holy painkillers were rebalanced and oxycodone was slightly buffed to make it strong enough to block surgery pain messages. Now there's four holy painkillers: a paracetamol subtype, a tramadol subtype, an oxycodone subtype, and the special monomial superpainkiller that blocks everything.
- Tessellates lost one litany (the AoE heal that used 150 hunger) and gained two: one that gives the holy oxycodone and one that injects the person in front of you with a variety of minor healing chems for an improved stabilizer.
- All AoE heals lost the requirement that the patient can hear, they're now actually useful for mascas situations rather than *only* mid-combat heals
- Altars now have a 10 minute cooldown between requesting items. This cooldown is on the altar, lasts 10 minutes, and applies to Request Cruciform, Prepare the Way, and Litany of Wholeness
- Factorial Omnitools scale at the same pace as Psionic ones (though they lack the medical applications)
- Reworks Church disks to be simpler code-wise and have distinct names
- Moved candles from Church vendor to Games vendor, as Church is now against crayon magic and we don't want them controlling the candles
- Adds a new silver subtype of wall obelisks that are weaker and are now placed around areas that either have primarily mechanical use (Scrap beacon, junk teleporter) or easy to bait maint mobs into (dorms sitting area)
- Adds public holy doors that open for anyone with a cruciform, places them in the Church Commons
- Fixes some odd areas in the Church
- Made Church able to print Fractal Rosaries and updated the description of those rosaries to fit lore.

/:cl:

### CODE REVIEW

File: code/modules/core_implant/cruciform/rituals/group.dm [hunk 1]
```diff
@@ -167,15 +163,16 @@
...
-		"Frase sexta.",
-		"Frase septima."
+		"Loquere filiis Israel, ut tollant mihi primitias: ab omni homine qui offeret ultroneus, accipietis eas.",
+		"Haec sunt autem quae accipere debeatis: aurum, et argentum, et aes.",
+		"hyacinthum et purpuram, coccumque bis tinctum, et byssum, pilos caprarum.",
+		"et pelles arietum rubricatas, pellesque janthinas, et ligna setim.",
+		"oleum ad luminaria concinnanda: aromata in unguentum, et thymiamata boni odoris.",
+		"lapides onychinos, et gemmas ad ornandum ephod, ac rationale.",
+		"Facientque mihi sanctuarium, et habitabo in medio eorum.",
+		"Amen"
```
Comment (Trilbyspaceclone) [inline code/modules/core_implant/cruciform/rituals/group.dm:175]:
```suggestion
		"Hyacinthum et purpuram, coccumque bis tinctum, et byssum, pilos caprarum.",
		"Et pelles arietum rubricatas, pellesque janthinas, et ligna setim.",
		"Oleum ad luminaria concinnanda: aromata in unguentum, et thymiamata boni odoris.",
		"Lapides onychinos, et gemmas ad ornandum ephod, ac rationale.",
		"Facientque mihi sanctuarium, et habitabo in medio eorum.",
		"Amen."
```

File: code/modules/core_implant/cruciform/rituals/base.dm [hunk 2]
```diff
@@ -326,7 +327,7 @@
 
 /datum/ritual/cruciform/base/recharge_others
 	name = "Empower"
-	phrase = "Potestas fidei communicanda est."
+	phrase = " Alter alterius onera portate." //"Carry each other’s burdens"
```
Comment (Trilbyspaceclone) [inline code/modules/core_implant/cruciform/rituals/base.dm:330]:
```suggestion
	phrase = "Alter alterius onera portate." //"Carry each other’s burdens"
```

**Comment by benj8560** [comment]
As a suggestion at some point not long ago Eris massively reduced the do after time for reading litanies from the book. I'd say by about half across the board. I think such a change would be nice to see here as well.

**Comment by SingingSpock** [comment]
Updated the comment with a lot of the stuff I've done.

File: code/modules/core_implant/cruciform/rituals/base.dm [hunk 3]
```diff
@@ -127,14 +127,14 @@
...
-		if((istype(CI) && CI.get_module(CRUCIFORM_CLERGY)) || prob(50))
+		if((istype(CI)))
 			to_chat(target, SPAN_DANGER("[H], faithful cruciform follower, cries for salvation at [t.name]!"))
 	return TRUE
 
 /datum/ritual/cruciform/base/reveal
 	name = "Reveal Adversaries"
 	phrase = "Et fumus tormentorum eorum ascendet in saecula saeculorum: nec habent requiem die ac nocte, qui adoraverunt bestiam, et imaginem ejus, et si quis acceperit caracterem nominis ejus." //"And the smoke of their torment will rise for ever and ever. There will be no rest day or night for those who worship the beast and its image, or for anyone who receives the mark of its name."
-	desc = "Gain knowledge of your surroundings to reveal evil in people and places. This can tell you about hostile creatures around you, rarely can help you spot traps and sometimes let you sense a monster disguised as a person."
+	desc = "Gain knowledge of your surroundings to reveal evil in people and places. This litany is imprecise, but can warn you of danger you donn't see."
```
Comment (benj8560) [inline code/modules/core_implant/cruciform/rituals/base.dm:104]: smaaaallll typo there. ''donn't see''

Comment (SingingSpock) [reply code/modules/core_implant/cruciform/rituals/base.dm:104]: Got it

**Comment by SingingSpock** [comment]
Okay, this is ready for review and probably testmerging. I tested a lot, but there's so many little changes that it's entirely possible something slipped through the cracks.

**Comment by SingingSpock** [comment]
Inquisitor spawner now works. Doctrine can be enforced, Primes can be demoted.

**Comment by SingingSpock** [comment]
Making a comment for this one since it's big and potentially controversial: Obelisks have been optimized in a few ways, which makes their process less laggy and thus improves their fire rate. However, they also now only activate for people who are visible to the obelisk and will only target things visible to the obelisk. No more sitting in one room while an obelisk in a second room kills all your enemies in a third room.

File: code/modules/core_implant/cruciform/machinery/wall_obelisk.dm [hunk 4]
```diff
@@ -17,7 +17,7 @@
 	var/active = TRUE //Were always active
 
 	var/area_radius = 7
-	var/damage = 120
+	var/damage = 40 //120 damage is insane, but these are a rarer resource (for now at least). Let's try 40
```
Comment (Trilbyspaceclone) [inline code/modules/core_implant/cruciform/machinery/wall_obelisk.dm:20]: Im greatly against this nerf.
these are located in areas like dorms, cryo spawn, and other RP related areas that should get some protection.

Comment (SingingSpock) [reply code/modules/core_implant/cruciform/machinery/wall_obelisk.dm:20]: The problem with that is that they also protect gamer areas like the scrap beacon. What if I tripled their starting ammo to make up for the thirding of their damage? Then I could in a separate PR later make another subtype that's got less ammo for the places where they're providing mechanical advantage, making those areas more reliant on Church upkeep to remain roachproof. (I don't want to touch mapping in this PR.)

Comment (SingingSpock) [reply code/modules/core_implant/cruciform/machinery/wall_obelisk.dm:20]: It's also worth noting that 40 is still enough to one-shot gib most roaches, it's only spiders and termites that this has a significant effect.

Comment (Trilbyspaceclone) [reply code/modules/core_implant/cruciform/machinery/wall_obelisk.dm:20]: Scrap beacon is a none factor, 40 and 120 both kill the roaches that are those lockers. its affects have no difference.
The only difference this change makes is affecting areas that *should be protected* less so, rather then gamer areas.

**Comment by SingingSpock** [comment]
Undid my changes to the current wall obelisks and made a silver subtype with a palette-swapped sprite that I'll map into the areas that shouldn't have superturrets (in a separate PR after eventual fullmerge because I don't want to lock down the map files)

**Comment by SingingSpock** [comment]
image
Image with the silver wall obelisks

**Comment by ChefDoggo** [comment]
the mega church PR is real

**Comment by SingingSpock** [comment]
It wasn't originally intended to be a mega PR, but at least it's stayed within its scope of Church code and not become a hodgepodge.

File: code/game/objects/structures/burrows.dm [hunk 5]
```diff
@@ -136,6 +132,13 @@
 		populated_burrows -= src
 		unpopulated_burrows |= src
 
+obj/structure/burrow/proc/obelisk_around()
```
Comment (Trilbyspaceclone) [inline code/game/objects/structures/burrows.dm:135]:
```suggestion
/obj/structure/burrow/proc/obelisk_around()
```

File: maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm [hunk 6]
```diff
@@ -50722,6 +50728,8 @@
 	},
 /turf/simulated/floor/tiled/white/monofloor,
 /area/nadezhda/medical/surgery)
+"kRB" = (
```
Comment (Trilbyspaceclone) [inline maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm:50731]: ?

Comment (Trilbyspaceclone) [reply maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm:50731]: theirs just a blank key?

Comment (SingingSpock) [reply maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm]: I don't know why it occurred, but I found and fixed it

**Comment by SingingSpock** [comment]
PR is nearing completion, so the changelog up there has been updated to include all the significant changes I could think of.

File: code/modules/core_implant/cruciform/rituals/path.dm [hunk 7]
```diff
@@ -719,90 +651,158 @@
...
 	cooldown = TRUE
 	cooldown_time = 10 MINUTES
 	cooldown_category = "repair"
-	power = 35
-	nutri_cost = 150
-	blood_cost = 80 //Aheal but not AOE, so little less bad
+	power = 50
+
+/datum/ritual/cruciform/factorial/self_repair/perform(mob/living/carbon/human/user, obj/item/implant/core_implant/cruciform/C)
+	var/debuff_amount = -5
```
Comment (Trilbyspaceclone) [inline code/modules/core_implant/cruciform/rituals/path.dm:663]: these should be moved to the self_repair path itself for editing via debug/admins

**Pull request merged**